src/components/picker/__tests__/index.spec.tsx

```diff
@@ -100,122 +97,185 @@ describe('Picker', () => {
+      //   jest.useFakeTimers();
+      //   expect(driver.isOpen()).toBeTruthy();
+      //   expect(onShow).toHaveBeenCalled();
+      // });
+
+      describe('Test Modal TopBar', () => {
+        it('should close and select items when pressing on cancel button', () => {
+          const driver = getDriver({mode: 'MULTI', topBarProps: {cancelLabel: 'Cancel'}});
+          driver.open();
+          expect(driver.isOpen()).toBeTruthy();
+          driver.selectItem(countries[2].label);
+          driver.selectItem(countries[4].label);
+          driver.done();
+          expect(driver.isOpen()).toBeFalsy();
+          expect(driver.getValue()).toEqual(`${countries[2].label}, ${countries[4].label}`);
+        });
+
+        it('should close without selecting items when pressing on cancel button', () => {
+          const driver = getDriver({mode: 'MULTI', topBarProps: {cancelLabel: 'Cancel'}});
+          driver.open();
+          expect(driver.isOpen()).toBeTruthy();
+          driver.selectItem(countries[2].label);
+          driver.selectItem(countries[4].label);
+          driver.cancel();
+          expect(driver.getValue()).toEqual(``);
+          expect(driver.isOpen()).toBeFalsy();
+        });
+      });
     });
   });
 
-  // TODO: this is a work in progress, the tests are not passing yet
-  describe.skip('Dialog', () => {
+  describe('Dialog', () => {
+    const dialogProps = {useDialog: true, customPickerProps: {migrateDialog: true}};
     describe('Test value', () => {
       it('Get correct value of a single item', () => {
         const driver = getDriver({
-          useDialog: true,
-          customPickerProps: {migrateDialog: true},
+          ...dialogProps,
           value: countries[2].value
         });
         expect(driver.getValue()).toEqual(countries[2].label);
       });
 
       it('Get correct value of multiple selected items', () => {
         const driver = getDriver({
-          useDialog: true,
-          customPickerProps: {migrateDialog: true},
+          ...dialogProps,
           value: [countries[2].value, countries[4].value]
         });
         expect(driver.getValue()).toEqual(`${countries[2].label}, ${countries[4].label}`);
       });
     });
 
-    describe('Test open', () => {
+    describe('Test open/close', () => {
       it('Should open when enabled', async () => {
-        const driver = getDriver({useDialog: true, customPickerProps: {migrateDialog: true}});
-
+        const driver = getDriver(dialogProps);
         expect(driver.isOpen()).toBeFalsy();
-        // driver.open();
-        // expect(driver.isOpen()).toBeTruthy();
         jest.useFakeTimers();
         act(() => driver.open());
         jest.runAllTimers();
-        // advanceAnimationByTime(10000);
-        // await new Promise(r => setTimeout(r, 1000));
-        await waitFor(() => expect(driver.isOpen()).toBeTruthy());
+        await waitFor(() => expect(driver.isOpen(dialogProps.useDialog)).toBeTruthy());
```

Comment:
I don't like the fact that we need to send the prop `useDialog` here.
We can take it to the other devs.

Reply:
Added to the next dev sync.
I'll ask Ethan about it tomorrow in our private sync.